tensorflow/core/common_runtime/gpu/gpu_cudamallocasync_allocator.cc

```diff
@@ -374,4 +359,40 @@ bool GpuCudaMallocAsyncAllocator::ClearStats() {
   return true;
 }
 
+void GpuCudaMallocAsyncAllocator::SetStream(void* stream) {
```

Comment:
The function is now misleadingly named -- maybe call it `SetStreamAndPreallocateMemory`?

Reply:
I rebased the PR and changed the function name.
I'm not sure I like it better then the short name.

Note, this PR depends on https://github.com/tensorflow/tensorflow/pull/50961 to be merged first. So do not merge it now. 
It is great to have it pre-reviewed to speed things up.